dali/python/nvidia/dali/pipeline.py

```diff
@@ -550,13 +550,20 @@ def contains_nested_datanode(nested):
         self._setup_input_callbacks()
         self._py_graph_built = True
 
+    def _setup_pipe_pool_dependency(self):
+        if self._py_pool_started:
+            # Pipeline backend doesn't really do anything with the pool, sole point of this call
+            # is to ensure lifetime of the pool exceeds the lifetime of the pipeline's backend
+            # so that shared memory managed by the pool is not freed before pipline is garbage collected.
+            # Otherwise pipline may try to access freed memory which leads to crashes at the Python teardown
```

Comment:
```suggestion
            # The sole point of this call is to ensure lifetime of the pool exceeds the lifetime of the pipeline's backend,
            # which runs external source operator that may access memory owned by the pool,
            # so that shared memory managed by the pool is not freed before pipline is garbage collected.
            # Otherwise pipline may try to access freed memory which leads to crashes at the Python teardown
```

Reply:
Adjusted the sentence, please take a look as I've removed I think unnecessary "which" parts.